atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/RetryableCassandraRequest.java

```diff
@@ -80,6 +81,10 @@ public void registerException(Exception exception) {
     }
 
     public AtlasDbDependencyException throwLimitReached() {
-        throw new RetryLimitReachedException(encounteredExceptions);
+        throw new RetryLimitReachedException(
+                encounteredExceptions,
+                EntryStream.of(triedHosts)
+                        .mapKeys(CassandraServer::cassandraHostName)
+                        .toMap());
```

Comment:
This code assumes that Cassandra server host names are unique, which is not always true. This caused errors in production.

I've put up a fix in https://github.com/palantir/atlasdb/pull/7085.